### PR TITLE
Add software tests and CI configuration

### PR DESCRIPTION
**Pull request by amotl**
Dear Frederic,

this patch will add some software tests and a CI/GHA configuration, it works well already [^1]. I believe it will tremendously help future maintenance.

With kind regards,
Andreas.

[^1]: https://github.com/cicerops/check_synology/actions/runs/2777733734

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	@test -e $(python) || python3 -m venv $(venv) || python -m venv $(venv)
+
+format: setup-virtualenv
+	$(pip) install --requirement=requirements-utils.txt
+	$(black) .
+	$(isort) .
+
+lint: setup-virtualenv
+	$(pip) install --requirement=requirements-utils.txt
+	$(flake8) --exit-zero *.py
```
Comment (amotl) [inline Makefile:21]: With `--exit-zero`, it will currently not bail out on any findings.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	$(MAKE) proselint
+
+proselint:
+	$(proselint) *.md || true
```
Comment (amotl) [inline Makefile:25]: Same here. Let's start in relaxed mode.

**Comment by amotl** [comment]
Hi Frederic,

I think this will be good to go. While the few test cases only bring in a code coverage of about 10%, at least the infrastructure will be in place. I will submit another patch which exercises the code base more thoroughly [^1].

With kind regards,
Andreas.

[^1]: May take some time, have to get my machine fixed beforehand and vacation is near.

**Comment by amotl** [comment]
Hi Frederic,

it may take a few more days to get my machine fixed, so I think all other work on this topic can be submitted with subsequent patches. In other words: Feel free to merge, in order to get the test infrastructure in place.

With kind regards,
Andreas.

**Comment by amotl** [comment]
Hi Frederic,

my machine got fixed quickly, so I will be able to continue to work a bit on the test harness. Do you have any objections against merging this PR as a baseline beforehand? I will submit subsequent work as separate PRs.

With kind regards,
Andreas.

**Comment by amotl** [comment]
The followup patch will be https://github.com/cicerops/check_synology/compare/add-ci...more-tests, bumping the code coverage from 10% to 27% by exercising the `load` mode for real.

**Comment by wernerfred** [comment]
Thx for that contribution. Sadly i'm super short on time in the last weeks - I promise i did not forget this addition but i would like to go through everything in detail before merging - and this needs time

**Comment by amotl** [comment]
Dear Frederic,

thank you. I remembered this patch yesterday as well, sweet that we apparently both did at about the same time ;]. Please take your time to review the patch properly.

With kind regards,
Andreas.

**Comment by amotl** [comment]
Friendly bump ;]. As a few reports about potential edge cases are arriving on the issue tracker, and also new contributions like #9, #41 and #46 (thanks a stack, @dommi22m and @bigitag!), I think it would be good to have a test harness in place.
